src/stores/DAOEntityStore.js

```diff
@@ -0,0 +1,68 @@
+// @flow
+
+import type RootStore from './RootStore';
+
+import {
+  action,
+  computed,
+  createTransformer,
+  observable,
+  runInAction,
+} from 'mobx';
+
+class DAOEntityStore<TEntity, TEntityMutator> {
+  _dao: DAO<TEntity, TEntityMutator>;
+  store: RootStore;
+
+  @observable entityItemsByID: Array<TEntity> = new Map();
+
+  constructor(rootStore: RootStore, dao: DAO<TEntity, TEntityMutator>) {
+    this._dao = dao;
+    this.store = rootStore;
+  }
+
+  //todo map another actions
+  @action
+  fetchByID = async (id: string, queryOptions?: QueryOptions): Promise<*> => {
+    const location = await this._doDAORequest('fetchByID', id, queryOptions);
+    if (!location) {
+      return;
+    }
+    // todo add babel plugin to autobind async stuff to runInAction
+    runInAction(() => {
+      this.entityItemsByID.set(location.id, location);
+    });
+  };
+
+  @computed
+  get all(): Array<TEntity> {
+    return this.entityItemsByID.values();
+  }
+
+  getByID = createTransformer((id: string): ?TEntity =>
+    this.entityItemsByID.get(id),
+  );
+
+  getByQueryFilters = createTransformer((queryFitlers: QueryFilters): Array<
+    TEntity,
+  > =>
+    this.entityItemsByID
+      .values()
+      .filter((entityItem: TEntity): boolean =>
+        DAOApi.doesSatisfyToQueryFilters(item, queryFilters),
+      ),
+  );
+
+  _doDAORequest = async (methodName, ...args): Promise<*> => {
```

Comment:
with this method I'm trying to achieve some kind of middleware like we have with redux: requestAction -> errorAction / successAction... there are no such actions here... but whenever we get an error, I save it to global ErrorStore, where the error can be visible for anyone who is interested in( for example logout , or snackbar message)

Reply:
Yeah, don't worry about solving all the problems now.  You need to understand when and how you'd actually use the error messages before you need to think about that